src/algorithms/pathfinders/dijkstra.ts

```diff
@@ -0,0 +1,111 @@
+import EdgeCost from '../models/edgecost';
+import Graph from '../models/graph';
+import Vertex from '../models/vertex';
+/**
+ * Implements Dijkstra's algorithm to find the shortest path between 2 vertices
+ */
+export default class DijkstraPathfinder {
+    /**
+     * Stores best known distances from starting vertex to all the others
+     */
+    distances: Map<number, {
+        distance: number;
+        sourceID: number;
+    }>;
+
+    /**
+     * stores all unvisited vertices
+     */
+    unvisited: Set<number>;
+
+    /**
+     * Returns an array of vertex IDs that represents the shortest path between 2 vertices
+     * @param startID : ID of starting vertex
+     * @param endID : ID of destination vertex
+     * @param mygraph : relevant graph
+     */
+    FindPath(startID: number, endID: number, mygraph: Graph): number[] {
+        this.Initialize(mygraph);
+        this.distances.set(startID, {distance: 0, sourceID: -9001});
+        this.unvisited.delete(startID);
+        let currentVertex = mygraph.getVertex(startID);
+        while (this.unvisited.size > 0) {

```

Comment:
Going through the whole set is extremely slow and this is not how dijkstra works. Imagine finding a path from Aalborg to Aarhus, why would you even consider paths through Copenhagen...
As I previously mentioned, dijkstra uses priority queue, which is not just significantly faster, but also needs more less memory (although in the worst case scenario, it could be as big as the whole graph 🙂)